File: csv-query/typescript/src/CsvTable.ts
import type { Row } from './Row.js';
import { UnknownColumnError } from './UnknownColumnError.js';

export class CsvTable {
  readonly headers: readonly string[];
  readonly rows: readonly Row[];

  constructor(headers: readonly string[], rows: readonly Row[]) {
    this.headers = headers;
    this.rows = rows;
  }

  validateColumn(column: string): void {
    if (!this.headers.includes(column)) {
      throw new UnknownColumnError(column);
    }
  }
}

File: csv-query/python/src/csv_query/csv_table.py
from __future__ import annotations

from .exceptions import UnknownColumnError
from .row import Row


class CsvTable:
    def __init__(self, headers: list[str], rows: list[Row]) -> None:
        self.headers = headers
        self.rows = list(rows)

    def validate_column(self, column: str) -> None:
        if column not in self.headers:
            raise UnknownColumnError(column)
